src/DevBetterWeb.Infrastructure/Services/WebhookHandlerService.cs

```diff
@@ -87,29 +98,77 @@ public async Task HandleNewCustomerSubscriptionAsync(string json)
+
+          if (string.IsNullOrEmpty(email))
+          {
+            throw new InvalidEmailException();
+          }
+
+          Invitation invite = await _newMemberService.CreateInvitationAsync(email, subscriptionId);
+
+          var webhookMessage = $"A new customer with email {email} has subscribed to DevBetter. They will be receiving a registration email.";
+          await _webhook.Send($"Webhook:\n{webhookMessage}");
+
+          await _newMemberService.SendRegistrationEmailAsync(invite);
+
+          await AddNewSubscriberBillingActivity(subscriptionId, email, paymentAmount);
         }
+      }
+    }
 
-        Invitation invite = await _newMemberService.CreateInvitationAsync(email, subscriptionId);
+    private async Task AddNewSubscriberBillingActivity(string subscriptionId, string email, decimal paymentAmount)
+    {
+      var subscriptionPlanName = _paymentHandlerSubscription.GetAssociatedProductName(subscriptionId);
+      var billingPeriod = _paymentHandlerSubscription.GetBillingPeriod(subscriptionId);
+      await _memberAddBillingActivityService.AddSubscriptionCreationBillingActivity(email, paymentAmount, subscriptionPlanName, billingPeriod);
+    }
 
-        var webhookMessage = $"A new customer with email {email} has subscribed to DevBetter. They will be receiving a registration email.";
-        await _webhook.Send($"Webhook:\n{webhookMessage}");
+    // Below two methods are for handling migration from Launchpass to Stripe
+    private async Task<bool> NewCustomerSubscriptionWithEmailAlreadyMember(string subscriptionId)
+    {
+      var customerId = _paymentHandlerSubscription.GetCustomerId(subscriptionId);
+      var email = _paymentHandlerCustomer.GetCustomerEmail(customerId);
 
-        await _newMemberService.SendRegistrationEmailAsync(invite);
+      var userIsMember = await _userLookupService.FindUserIsMemberByEmailAsync(email);
 
-        var subscriptionPlanName = _paymentHandlerSubscription.GetAssociatedProductName(subscriptionId);
-        var billingPeriod = _paymentHandlerSubscription.GetBillingPeriod(subscriptionId);
-        await _memberAddBillingActivityService.AddSubscriptionCreationBillingActivity(email, paymentAmount, subscriptionPlanName, billingPeriod);
+      if (userIsMember)
+      {
+        return true;
       }
+
+      return false;
+
+    }
+
+    private async Task HandleNewCustomerSubscriptionWithEmailAlreadyMember(string subscriptionId, decimal paymentAmount)
+    {
+      var customerId = _paymentHandlerSubscription.GetCustomerId(subscriptionId);
+      var email = _paymentHandlerCustomer.GetCustomerEmail(customerId);
+
+      await AddNewSubscriberBillingActivity(subscriptionId, email, paymentAmount);
+
+      var subscriptionDateTimeRange = _paymentHandlerSubscription.GetDateTimeRange(subscriptionId);
+
+      var userId = await _userLookupService.FindUserIdByEmailAsync(email);
+      var spec = new MemberByUserIdSpec(userId);
```

Comment:
Is there no `MemberByEmailSpec`? If not, would it make sense to try and add one so you don't need the double lookup here to get a userId by email and then a member by userid?

Reply:
The problem is, we don't store email on member, so conceptually I feel like it's weird to do that? Even though it'll always be 1:1, so I guess it makes sense. I think this'd start to get complicated because you'd have the user lookup service involved. But perhaps adding a method/service somewhere to do this would probably be the way to go. It's certainly a useful functionality. Since I think it'd be somewhat complicated, I'll add an issue.